athena-db2-as400/src/main/java/com/amazonaws/athena/connectors/db2as400/Db2As400MetadataHandler.java

```diff
@@ -0,0 +1,481 @@
+        SpillLocation spillLocation = makeSpillLocation(getSplitsRequest);
+        Split.Builder splitBuilder;
+        Set<Split> splits = new HashSet<>();
+
+        LOGGER.debug("partitionContd: {}", partitionContd);
+
+        for (int curPartition = partitionContd; curPartition < partitions.getRowCount(); curPartition++) {
+            FieldReader locationReader = partitions.getFieldReader(PARTITION_NUMBER);
+            locationReader.setPosition(curPartition);
+            String partInfo = String.valueOf(locationReader.readText());
+
+            LOGGER.debug("{}: Input partition is {}", getSplitsRequest.getQueryId(), partInfo);
+
+            // We are passing partition column and partition number for each split(),
+            // so that it can be used while framing the query where clause in
+            // QueryStringBuilder as partition identification.
+            // Included partition information to split if the table is partitioned
+            if (partInfo.contains(":::")) {
+                String[] partInfoAr = partInfo.split(":::");
+                splitBuilder = Split.newBuilder(spillLocation, makeEncryptionKey())
+                        .add(PARTITIONING_COLUMN, partInfoAr[0])
+                        .add(PARTITION_NUMBER, partInfoAr[1]);
+            }
+            else {
+                splitBuilder = Split.newBuilder(spillLocation, makeEncryptionKey())
+                        .add(PARTITION_NUMBER, partInfo);
+            }
+            splits.add(splitBuilder.build());
+            if (splits.size() >= MAX_SPLITS_PER_REQUEST) {
+                //We exceeded the number of split we want to return in a single request, return and provide a continuation token.
+                return new GetSplitsResponse(getSplitsRequest.getCatalogName(), splits, encodeContinuationToken(curPartition));
+            }
+        }
+        return new GetSplitsResponse(getSplitsRequest.getCatalogName(), splits, null);
+    }
+
+    /**
+     * If the table have partitions fetch those partition details from DB2 metadata tables.
+     * This information will be used while forming the custom query to get specific partition as a split.
+     *
+     * @param parameters
+     * @throws Exception
+     */
+    private String getColumnName(List<String> parameters) throws Exception
+    {
+        try (Connection connection = getJdbcConnectionFactory().getConnection(getCredentialProvider());
+             PreparedStatement preparedStatement = new PreparedStatementBuilder().withConnection(connection).withQuery(Db2As400Constants.COLUMN_INFO_QUERY).withParameters(parameters).build();
+             ResultSet resultSet = preparedStatement.executeQuery()) {
+            if (resultSet.next()) {
+                return resultSet.getString("COLUMN_NAME");
+            }
+            else {
+                LOGGER.error("Column Name not found in the table {} ", parameters.get(1));
+                return null;
```

Comment:
In my comment I mentioned that we should just throw after logging an error: https://github.com/awslabs/aws-athena-query-federation/pull/1245#discussion_r1192924867

Unless there's a reason for returning null here?

Reply:
Hi Henry, addressed.Thank you